python/src/scipp/compat/mantid.py

```diff
@@ -442,26 +449,19 @@ def from_mantid(workspace, **kwargs):
         raise RuntimeError('Unsupported workspace type {}'.format(
             workspace.id()))
 
-    try:
-        # TODO Is there ever a case where a Workspace2D has a separate monitor
-        # workspace? This is not handled by ExtractMonitors above, I think.
-        if monitor_ws is None:
-            monitor_ws = workspace.getMonitorWorkspace()
-        if monitor_ws.id() == 'Workspace2D':
-            converter = convert_Workspace2D_to_dataarray
-        elif monitor_ws.id() == 'EventWorkspace':
-            converter = convertEventWorkspace_to_dataarray
-
-        monitors = convert_monitors_ws(monitor_ws, converter, **kwargs)
-        for name, monitor in monitors:
-            # Remove redundant information that is duplicated from workspace
-            del monitor.labels['sample_position']
-            del monitor.labels['detector_info']
-            del monitor.attrs['run']
-            del monitor.attrs['sample']
-            dataset.attrs[name] = sc.Variable(value=monitor)
-    except RuntimeError:
-        pass
+    # TODO Is there ever a case where a Workspace2D has a separate monitor
+    # workspace? This is not handled by ExtractMonitors above, I think.
+    if monitor_ws is None:
+        monitor_ws = workspace.getMonitorWorkspace()
+    if monitor_ws.id() == 'Workspace2D':
+        converter = convert_Workspace2D_to_dataarray
+    elif monitor_ws.id() == 'EventWorkspace':
+        converter = convertEventWorkspace_to_dataarray
+    
+    monitors = convert_monitors_ws(monitor_ws, converter, **kwargs)
```

Comment:
Could it be that `monitor_ws` is still `None` here? Was that the reasons for the old `try` block? Have you tested this

Reply:
For example for a `TableWorkspace`?

Reply:
You're right, that would fall through at present.